service-runner.js

```diff
@@ -178,19 +204,35 @@ ServiceRunner.prototype._runMaster = function() {
     // Set up rolling restarts
     process.on('SIGHUP', this._rollingRestart.bind(this));
 
-    return this._startWorkers(this.config.num_workers);
+    return this._startWorkers(this.config.num_workers)
+    .then(function(workers) {
+        setTimeout(function() {
```

Comment:
The worker could send a heartbeat right before starting the loading of the service modules, couldn't it?